.tekton/devfile-sample-go-basic-pull-request.yaml

```diff
@@ -201,10 +202,10 @@ spec:
       workspaces:
       - name: source
         workspace: workspace
-    - name: build-container
+    - name: build-container-amd64
       params:
```

Comment:
This task doesn't have an `ARCH` parameter. Is this intentional?

Reply:
This is the standard 'buildah' task, it builds the amd64 image the same way all other images are currently built.

Reply:
Ack. I'm just thinking from the verification side, I need to know all the arches a pipeline is expected to produce. I'd have to, somehow, identify all the tasks that are build tasks and for the ones without the `ARCH` parameter default to a hard-coded value, i.e. `amd64`. 


Also, using different task definitions for different architectures should be allowed, but I can imagine this raising eyebrows. Ideally, the same task is used for all arches to ensure some sort of consistency. I expect EC will be asked to write a policy rule that ensures this for Red Hat products, for example.

fyi @arewm

Reply:
Would this be resolved if a bundle or git resolver were used? Then the "same" content can be contained in the original location and just the name of each task modified for each architecture.

I don't think we should be concerned about consistency between single-arch and multi-arch builds (i.e. it is okay to have suffixes on all task names).

Reply:
We don't want to do amd64 builds on VMs though, so if we were going to try and have it all the same task it would be kinda yuck, you would basically have two tasks merged into one with a giant if statement separating them.

Reply:
> We don't want to do amd64 builds on VMs though

That drives to the core of what I was trying to get to. If this is true for RHTAP multi-arch builds, then the verification is a little more elaborate.

From a different perspective, I think having a completely different build step for just one of the platforms is going to introduce all sorts of unexpected behavior.

To clarify, I'm saying that if a pipeline is building a single arch image, then that image can be built natively in OpenShift (probably makes the most sense), or on a remote host (if someone really wants this for whatever reason). And when a pipeline is building a multi-arch image, then each arch of that image should be built on a corresponding remote host.

Reply:
There is no reason why we can't do that, especially if we are using dynamic provisioning. Basically you just need to modify the aws config at https://github.com/redhat-appstudio/multi-arch-controller/blob/main/deploy/overlays/dev-template/host-config.yaml#L10 and add appropriate config to start an arm64 machine.